sematic/resolvers/local_resolver.py

```diff
@@ -295,7 +295,7 @@ def _make_cache_namespace(self) -> Optional[str]:
     def _resolution_did_cancel(self) -> None:
         super()._resolution_did_cancel()
         api_client.cancel_resolution(self._root_future.id)
-        self._sio_client.disconnect()
+        self._clean_up_resolution(save_graph=True)
```

Comment:
bug: we weren't necessarily cleaning up external resources if the SIGINT from the pod cancellation reached the resolver before the asyncio cancellation event did. This fixes it